core/tests/webdriverio_utils/ExplorationEditorMainTab.js

```diff
@@ -516,6 +516,8 @@ var ExplorationEditorMainTab = function() {
   // possible to click on them to view their contents, as clicks instead open
   // the rich text editor.
   this.expectContentToMatch = async function(richTextInstructions) {
+    await waitFor.visibilityOf(
+      stateContentDisplay, 'State content display not showing up');
```

Comment:
aren't we already waiting for stateContentDisplay visibility here: https://github.com/oppia/oppia/blob/0bfc01b0913d5a2cc16d305049d4ddc19265b62f/core/tests/webdriverio_utils/forms.js#L495

(Also, I think keeping the visibility check for general functions at the bottom level should be better as these functions can be reused without worrying about different checks.)

Reply:
Yeah, I agree. It's harder to pin down this one though, and in fact technically that "bottom" check was supposed to catch the extensions issue too, but didn't.

I figured that perhaps extensions is special due to the animation, and I think in this case there is a transition of the state editor as well. But it's hard to say whether this is fully fixing the issue or not. There isn't really much harm leaving it in, though. 

Do you think I should keep it or drop it? I'm happy either way, let me know your thoughts :)

Thanks!

Reply:
I'm fine with keeping it as it's somehow helping!

Reply:
Note: this guard did not work. https://github.com/oppia/oppia/actions/runs/7098449227/job/19321586522

I reopened #16097 and noted this there as well so that we remove it in the next iteration.